apps/site/assets/ts/schedule/components/line-diagram/graphics/Diagram.tsx

```diff
@@ -71,30 +71,93 @@ const branchingDescription = (stopTree: StopTree): string => {
 };
 
 const LiveVehicleIconSet = ({
-  stopTree,
-  stopId,
+  isStart,
+  stop,
   liveData
 }: {
-  stopTree: StopTree;
-  stopId: StopId;
+  isStart: boolean;
+  stop: RouteStop;
   liveData?: LiveDataByStop;
 }): ReactElement<HTMLElement> | null => {
+  const stopId = stop.id;
   if (!liveData || !liveData[stopId]) return null;
   const vehicles = uniqBy(liveData[stopId].vehicles, "id");
   // Hide vehicles arriving to the origin from 'off the line'
-  const vehicleData = isStartNode(stopTree, stopId)
+  const vehicleData = isStart
     ? vehicles.filter(vehicle => vehicle.status === "stopped")
     : vehicles;
 
   return (
     <VehicleIcons
       key={`${stopId}-vehicles`}
-      stop={stopForId(stopTree, stopId)}
+      stop={stop}
       vehicles={vehicleData}
     />
   );
 };
 
+interface SimpleProps {
+  routeStopList: RouteStop[];
+  route: Route;
+  directionId: DirectionId;
+  alerts: Alert[];
+  liveData?: LiveDataByStop;
+}
+
+const SimpleDiagram = ({
+  routeStopList,
+  route,
+  directionId,
+  alerts,
+  liveData
+}: SimpleProps): ReactElement<HTMLElement> => (
+  <>
+    {routeStopList.map((routeStop, index) => (
+      <LiveVehicleIconSet
+        // eslint-disable-next-line react/no-array-index-key
+        key={`${index}-${routeStop.id}`}
+        isStart={index === 0}
+        stop={routeStop}
+        liveData={liveData}
+      />
+    ))}
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      role="img"
+      aria-labelledby="diagram-title diagram-desc"
+      className={`line-diagram-svg ${routeToModeName(route)}`}
+      width={`${diagramWidth(1)}px`}
+      height="100%"
+      style={{ left: BASE_LINE_WIDTH / 2, overflow: "visible" }}
+    >
+      <title id="diagram-title">Line diagram for {routeName(route)}</title>
+      <desc id="diagram-desc">
+        {diagramDescription(routeStopList.length, route, directionId)}
+      </desc>
+      <defs>{DiagonalHatchPattern()}</defs>
+      {/* Draw lines between stops */
+      routeStopList.map((routeStop, index) => {
+        const nextStop = routeStopList[index + 1];
+        if (!nextStop) return null;
+        return (
+          <SimpleLine
+            key={`${routeStop.id}-${nextStop.id}`}
+            fromId={routeStop.id}
+            toId={nextStop.id}
+            alerts={alerts}
+          />
+        );
+      })}
+
+      {/* Draw circles for each stop */
+      routeStopList.map((routeStop, index) => (
+        // eslint-disable-next-line react/no-array-index-key
+        <Stop key={`${index}-${routeStop.id}`} stopId={routeStop.id} />
```

Comment:
We really shouldn't be using indexes as keys, for the underlying list items could change, but the index does not.  Is the id not unique here?  Is there some other way to get a unique id? (Like name-id-zone)

Reply:
The `routeStop.id` is not guaranteed to be unique amongst the list of stops. Because of this edge case where stops are repeated along the route. But maybe I can add a number to each item in the list and use that as the key...

Reply:
@kotva006 I've pushed an update to change this.